src/types.jl

```diff
@@ -48,53 +48,74 @@ function metadata(result::API.MYSQL_RES)
     return unsafe_wrap(Array, rawfields, nfields)
 end
 
-mutable struct Query{hasresult, names, T}
+# resulttype is a symbol with values :none :streaming :default
+# names and types relate to the returned columns in the query
+mutable struct Query{resulttype, names, T}
     result::Result
     ptr::Ptr{Ptr{Int8}}
     ncols::Int
     nrows::Int
 end
 
+# does not implement Table interface - lower level interaction
+mutable struct StreamingQuery{hasresult, names, T}
+    result::Result
+    ptr::Ptr{Ptr{Int8}}
+    ncols::Int
+end
+
 function julia_type(field_type, notnullable, isunsigned)
     T = API.julia_type(field_type)
     T2 = isunsigned ? unsigned(T) : T
     return notnullable ? T2 : Union{Missing, T2}
 end
 
 """
-    MySQL.Query(conn, sql, sink=Data.Table; kwargs...) => MySQL.Query
+    MySQL.Query(conn, sql; kwargs...) => MySQL.Query
 
 Execute an SQL statement and return a `MySQL.Query` object. Result rows can be
-iterated as NamedTuples via `Data.rows(query)` where `query` is the `MySQL.Query`
+iterated as NamedTuples via `Table.rows(query)` where `query` is the `MySQL.Query`
 object.
 
+Supported Key Word Arguments:
+* `streaming` - Defaults to false. If true, length of the result size is unknown as the result is returned row by row. May be more memory efficient.
+
 To materialize the results as a `DataFrame`, use `MySQL.query(conn, sql) |> DataFrame`.
 """
 function Query(conn::Connection, sql::String; kwargs...)
     conn.ptr == C_NULL && throw(MySQLInterfaceError("Method called with null connection."))
     MySQL.API.mysql_query(conn.ptr, sql) != 0 && throw(MySQLInternalError(conn))
-    result = MySQL.Result(MySQL.API.mysql_store_result(conn.ptr))
+
+    if get(kwargs, :streaming, false)

```

Comment:
it's more idiomatic to change the function signature to `function Query(conn::Connection, sql::String; stream::Bool=false, kwargs...)`, then for this if-block, you'd say `if streaming`